build/media-suite_compile.sh

```diff
@@ -2432,27 +2438,107 @@ if [[ $mpv != n ]] && pc_exists libavcodec libavformat libswscale libavfilter; t
+        # The alternative is having users only specify everything after it in the config, e.g.
+        # `lua=luajit` and `cdda=enabled` (optionally omitting `=enabled` for on/off features)
+        local meson_opts=()
+        local default_lib="both"
+        local option
+        for option in "${MPV_OPTS[@]}"; do
+            # Process Waf flags into Meson options.
+            # starting with boolean flags
+            if [[ $option =~ ^--enable-(gpl|cplayer|build-date|tests|ta-leak-report) ]]; then
+                meson_opts+=("${option/--enable-/-D}=true")
+            elif [[ $option =~ ^--disable-(gpl|cplayer|build-date|tests|ta-leak-report) ]]; then
+                meson_opts+=("${option/--disable-/-D}=false")
+            elif [[ $option =~ ^--enable-libmpv(-shared)? ]]; then
+                meson_opts+=("-Dlibmpv=true")
+            elif [[ $option =~ ^--disable-libmpv(-shared)? ]]; then
+                meson_opts+=("-Dlibmpv=false")
+                default_lib="static"
+            # features in the misc section
+            elif [[ $option =~ ^--enable-(cdda|cplugins|dvbin|dvdnav|iconv|javascript|lcms2|libarchive|libavdevice|libbluray|pthread-debug|rubberband|sdl2|sdl2-gamepad|stdatomic|uchardet|uwp|vapoursynth|vector|win32-internal-pthreads|zimg|zlib) ]]; then
+                meson_opts+=("${option/--enable-/-D}=enabled")
+            elif [[ $option =~ ^--disable-(cdda|cplugins|dvbin|dvdnav|iconv|javascript|lcms2|libarchive|libavdevice|libbluray|pthread-debug|rubberband|sdl2|sdl2-gamepad|stdatomic|uchardet|uwp|vapoursynth|vector|win32-internal-pthreads|zimg|zlib) ]]; then
+                meson_opts+=("${option/--disable-/-D}=disabled")
+            # Audio option time! This goes in twos; we're unconditionally disabling non-Windows options and warning the user.
+            elif [[ $option =~ ^--enable-(alsa|audiounit|coreaudio|oss-audio|pipewire|pulse|sndio) ]]; then
+                meson_opts+=("${option/--enable-/-D}=disabled")
+                do_simple_print "${orange}Disabling option '${option}' as it's not usable on Windows"'!'"${reset}"
+            elif [[ $option =~ ^--enable-(jack|openal|opensles|sdl2-audio|wasapi) ]]; then
+                meson_opts+=("${option/--enable-/-D}=enabled")
+            elif [[ $option =~ ^--disable-(jack|openal|opensles|sdl2-audio|wasapi) ]]; then
+                meson_opts+=("${option/--disable-/-D}=disabled")
+            # Video corner, same as with audio
+            elif [[ $option =~ ^--enable-(cocoa|drm|egl-android|*-drm|*-wayland|*-x11|*-cocoa|gbm|rpi|vdpau|vdpau-*|vaapi|vaapi-*|wayland|x11|xv) ]]; then
+                meson_opts+=("${option/--enable-/-D}=disabled")
+                do_simple_print "${orange}Disabling option '${option}' as it's not usable on Windows"'!'"${reset}"
+            elif [[ $option =~ ^--enable-(caca|d3d11|direct3d|egl|egl-*|plain-gl|gl|gl-*|jpeg|libplacebo|sdl2-video|shaderc|sixel|spirv-cross|vulkan) ]]; then
```

Comment:
There's also some things to note about SPIRV-Cross, see issue linked in the initial comment on the PR